[ext] feat: add a Rate Now button under the YT videos

sigmike opened the pull request:
Related issue: #413

Initial attempt to add a "Rate now" button on YouTube (after the previous work in https://github.com/tournesol-app/tournesol/pull/459).

For now the design is just a copy of the existing button:
![image](https://user-images.githubusercontent.com/28259/207121573-76b17f85-85a2-46dc-a3c0-5eb7d613a335.png)

With the iframe open:
![image](https://user-images.githubusercontent.com/28259/207123712-f296bbe5-f24c-4789-9d1f-a9bd51d8d008.png)


It works pretty well but there are some minor issues:
- The dialog is probably not tall enough
- When you're not logged in on tournesol.app it asks you to log in but when you've done that the dialog is closed and you have to click again on "Rate now". The behavior is the same on the "Rate later" button but it feels a little weirder here. It may be best to change that on both buttons, but it's not trivial.
- When you submit your comparison the dialog is not closed. It gives the opportunity to complete the comparison or to load another video, so I guess it's ok to keep that behavior. And fixing it may be tough (similar difficulty as the point above).

GresilleSiffle commented:
It looks great, I'm going to read the code today.

> The dialog is probably not tall enough

It's true. With a height of 800 pixels we can see the main criterion with all optional criteria at once.

> When you're not logged in on tournesol.app it asks you to log in but when you've done that the dialog is closed and you have to click again on "Rate now". The behavior is the same on the "Rate later" button but it feels a little weirder here. It may be best to change that on both buttons, but it's not trivial.

Yes, it's a known issue. This is something we should improve somehow.

> When you submit your comparison the dialog is not closed. It gives the opportunity to complete the comparison or to load another video, so I guess it's ok to keep that behavior. And fixing it may be tough (similar difficulty as the point above).

It looks OK for me, it allows us to compare the video several times which is good for Tournesol : ) We could also add a close button to make the close action more "actionable", hoping it will be more clear for users that they must close the iframe themselves (I have a draft mock-up of the close button on my laptop).

**Things I see to improve in the future (we can create issues after this PR or fix them directly here)**
- use a different icon for the rate later and rate now buttons
- we can imitate the YT buttons style with a yellow filled rounded background

GresilleSiffle reviewed (commented):
Thank you @sigmike 

The feature looks really good to me, except the following two points:
- iframe height that could be increased
- the buttons should have different icons (not sure if we want to it in this PR)

I'll let @amatissart and @lfaucon also have a look before merging.

sigmike commented on browser-extension/src/addRateButtons.js:
```diff
@@ -0,0 +1,144 @@
+  const timer = window.setInterval(createButtonIsReady, 300);
+
+  /**
+   * Create the Rate Later button.
```
Done in 6bd4e7a2.

sigmike commented on browser-extension/src/addRateButtons.js:
```diff
@@ -0,0 +1,144 @@
+      button.setAttribute('class', 'tournesol-rate-button');
+
+      // Image td for better vertical alignment
+      const img_td = document.createElement('td');
```
Changed in 62191c55.

sigmike commented:
I made the changes we discussed:
- moved the snackbar into the title bar when embedded
- changed the icons

In dark mode:
![image](https://user-images.githubusercontent.com/28259/207867559-08ffbee9-8e0d-4cd8-92c9-274cec9abf08.png)

The snackbar:
![image](https://user-images.githubusercontent.com/28259/207867824-1d87bd89-35a9-4bd9-b811-1abade9421bb.png)

GresilleSiffle reviewed (approved):
Very happy to see this new feature developed :+1:
